packages/core-data/src/entities.js

```diff
@@ -216,7 +216,7 @@ async function loadPostTypeEntities() {
 			baseURL: `/${ namespace }/${ postType.rest_base }`,
 			baseURLParams: { context: 'edit' },
 			name,
-			label: postType.labels.singular_name,
+			label: postType.name,
```

Comment:
This change is going to be difficult to test and the results could be unexpected (although likely relatively minor).

Ideally we would modify the Types API endpoint to provide access to the singular label as that will avoid this change cause knock on effects to UI labels (some of which may be handled in 3rd party plugins).